disnake/ui/action_row.py

```diff
@@ -24,86 +24,170 @@
 
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, List, Optional, Sequence, TypeVar, Union
-
-from ..components import ActionRow as ActionRowComponent, NestedComponent, SelectOption
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    ClassVar,
+    Generator,
+    Generic,
+    List,
+    Literal,
+    Optional,
+    Sequence,
+    Tuple,
+    TypeVar,
+    Union,
+    overload,
+)
+
+from ..components import (
+    ActionRow as ActionRowComponent,
+    Button as ButtonComponent,
+    NestedComponent,
+    SelectMenu as SelectComponent,
+    SelectOption,
+)
 from ..enums import ButtonStyle, ComponentType, TextInputStyle
 from ..utils import MISSING
 from .button import Button
 from .item import WrappedComponent
 from .select import Select
 from .text_input import TextInput
 
-__all__ = ("ActionRow",)
-
 if TYPE_CHECKING:
     from ..emoji import Emoji
+    from ..message import Message
     from ..partial_emoji import PartialEmoji
     from ..types.components import ActionRow as ActionRowPayload
 
-    ActionRowT = TypeVar("ActionRowT", bound="ActionRow")
-    Components = Union[
-        ActionRowT,
-        WrappedComponent,
-        Sequence[Union[ActionRowT, WrappedComponent, Sequence[WrappedComponent]]],
-    ]
+__all__ = (
+    "ActionRow",
+    "Components",
+    "MessageUIComponent",
+    "ModalUIComponent",
+    "MessageActionRow",
+    "ModalActionRow",
+)
+
 
+MessageUIComponent = Union[Button[Any], Select[Any]]
+ModalUIComponent = Union[TextInput, Select[Any]]
```

Comment:
Pretty sure these can use `None` instead of `Any`

Reply:
These could be None, yeah, but as-is we'd also support components from views. I don't see why anyone would put their view components inside an action row, but I don't think it'd break anything. I'm not sure if we should just disallow that by making it None.

Reply:
Ah yeah, fair enough. There isn't really a point in disallowing components with views.